codegen/core/src/main/java/software/amazon/smithy/python/codegen/ClientGenerator.java

```diff
@@ -326,17 +326,26 @@ async def _handle_execution(
                                     )
                                 except SmithyRetryException:
                                     raise context_with_response.response
+                                logger.debug(
+                                    "Retry needed. Attempting request #%s in %s seconds.",
+                                    retry_token.retry_count + 1,
+                                    retry_token.retry_delay
+                                )
                                 await sleep(retry_token.retry_delay)
                                 current_body =  context_with_transport_request.transport_request.body
                                 if (seek := getattr(current_body, "seek", None)) is not None:
                                     await seek(0)
                             else:
                                 # Step 8: Invoke record_success
+                                logger.debug(
+                                    "Attempt %s succeeded. Not retrying request.",
+                                    retry_token.retry_count + 1
+                                )
                                 retry_strategy.record_success(token=retry_token)
                                 break
                     except Exception as e:
                         if context.response is not None:
-                            logger.exception(f"Exception occurred while handling: {context.response}")
+                            logger.exception("Exception occurred while handling: %s", context.response)
```

Comment:
What gets logged with this? Do we have an example response payload?

Reply:
Yup, here is an example response:
```
ERROR:bedrock_runtime.client:Exception occurred while handling: Missing Authentication Token
```